src/router/router.ts

```diff
@@ -0,0 +1,316 @@
+import { InjectionKey } from 'vue'
+import { RouteComponent, RouteLocationRaw, RouteRecordRaw } from 'vue-router'
+import { createWorkspaceRoutes } from '@/router/routes'
+
+export type Route = Exclude<RouteLocationRaw, string>
+
+type WorkspaceRoutes = ReturnType<typeof createWorkspaceRoutes>
+type WorkspaceRouteKey = keyof WorkspaceRoutes
+type WorkspaceRoute = ReturnType<WorkspaceRoutes[WorkspaceRouteKey]>
+type WorkspaceNamedRoute = WorkspaceRoute['name']
+
+type WorkspaceRouteRecordParent = { name?: WorkspaceNamedRoute, children: WorkspaceRouteRecord[] }
+type WorkspaceRouteRecordChild = { name: WorkspaceNamedRoute }
+type WorkspaceRouteRecord = Omit<RouteRecordRaw, 'name' | 'children'> & WorkspaceRouteRecordParent | WorkspaceRouteRecordChild
+
+export const workspaceRoutesKey: InjectionKey<WorkspaceRoutes> = Symbol('WorkspaceRoutes')
+
+type WorkspaceComponent = () => Promise<RouteComponent>
+type WorkspaceRouteComponents = Record<WorkspaceRouteKey, WorkspaceComponent>
+
+export function createWorkspaceRouteRecords(components: Partial<WorkspaceRouteComponents>): WorkspaceRouteRecord[] {
+  return [
+    {
+      path: 'flow-runs',
+      meta: {
+        can: 'read:flow_run',
+      },
+      children: [
+        {
+          name: 'workspace.flow-runs',
+          path: '',
+          component: components.flowRuns,
+        },
+        {
+          name: 'workspace.flow-runs.flow-run',
+          path: 'flow-run/:flowRunId',
+          component: components.flowRun,
+        },
+        {
+          name: 'workspace.flow-runs.flow-run-radar',
+          path: 'flow-run/:flowRunId/radar',
+          component: components.flowRunRadar,
+        },
+        {
+          name: 'workspace.flow-runs.task-run',
+          path: 'task-run/:taskRunId',
+          component: components.taskRun,
+        },
+      ],
+    },
+    {
+      path: 'flows',
+      meta: {
+        can: 'read:flow',
+      },
+      children: [
+        {
+          name: 'workspace.flows',
+          path: '',
+          component: components.flows,
+        },
+        {
+          name: 'workspace.flows.flow',
+          path: 'flow/:flowId',
+          component: components.flow,
+        },
+        {
+          path: 'collections',
+          meta: {
+            can: 'access:collections',
```

Comment:
To confirm, collections is something somebody can always access if they have general read:flow permissions, right?

Reply:
yes (and the feature flag), it's actually just a "read-only" type feature that's more like marketing material than an actually impactful change to how things operate.